### PR TITLE
Allow passing addl. Guzzle middleware in to GuzzleClient in createDefaultClient().

### PR DESCRIPTION
**Pull request by garthbrantley**
Extend ShopifyClient to allow passing in array of middleware to be added to GuzzleClient handler stack.

This is preferable to passing in an entire GuzzleClient. The reason is because when you pass in a GuzzleClient from outside, you lose all the logic in the retry middleware. You also lose all the logic in the wrapRequestData Command-to-request transformer.

### CODE REVIEW

File: src/ShopifyClient.php [hunk 1]
```diff
@@ -568,13 +568,18 @@ private function validateConnectionOptions(array $connectionOptions)
...
      */
-    private function createDefaultClient(): GuzzleClient
+    private function createDefaultClient(array $guzzleMiddleware = []): GuzzleClient
     {
         $baseUri = 'https://' . str_replace('.myshopify.com', '', $this->connectionOptions['shop']) . '.myshopify.com';
 
         $handlerStack = HandlerStack::create(new CurlHandler());
         $handlerStack->push(Middleware::retry([$this, 'retryDecider'], [$this, 'retryDelay']));
 
+        foreach ($guzzleMiddleware as $curMiddleware)
```
Comment (bakura10) [inline src/ShopifyClient.php]: Can you please move the opening { to follow the same code conventions ? :)

**Comment by bakura10** [comment]
Wouldn't it be better to try to take the existing guzzle client that is passed and attached it the additional middlewares? I didn't check if the API of the client offers that but that sounds much cleaner than adding an extra parameter.

**Comment by garthbrantley** [comment]
@bakura10 I don't believe there is any way to add middleware to existing client. That is why I passed in the middleware stack.

See this:
https://github.com/guzzle/guzzle/issues/2237

File: src/ShopifyClient.php (hunk 1 above)
Comment (garthbrantley) [reply src/ShopifyClient.php]: Sure, sorry about that.

**Comment by bakura10** [comment]
Thanks! Last thing I forgot, could you please update the changelog (this can be version 6.2.1)

**Comment by garthbrantley** [comment]
Ok, I added a changelog entry for 6.2.1

**Comment by bakura10** [comment]
Thanks. Tagged as 6.2.1

**Pull request merged**